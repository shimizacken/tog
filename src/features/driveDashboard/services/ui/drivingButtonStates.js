export const DrivingButtonStates = {
    TURNED_OFF: 'TURNED_OFF',
    READY: 'READY',
    DRIVE: 'DRIVE',
    ERROR: 'ERROR'
};