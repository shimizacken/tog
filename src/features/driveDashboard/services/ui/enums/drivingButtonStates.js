const DrivingButtonStates = {
    TURNED_OFF: 'TURNED_OFF',
    READY: 'READY',
    DRIVE: 'DRIVE',
    ERROR: 'ERROR'
};

export default DrivingButtonStates;
