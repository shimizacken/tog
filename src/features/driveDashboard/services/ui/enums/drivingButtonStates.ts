export enum DrivingButtonStates {
  TURNED_OFF = "TURNED_OFF",
  READY = "READY",
  DRIVE = "DRIVE",
  ERROR = "ERROR",
  SWITCHING = "SWITCHING",
}
