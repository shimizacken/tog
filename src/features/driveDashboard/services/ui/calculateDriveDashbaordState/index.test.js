import calculateDrivingStatus from ".";
import { DrivingButtonStates } from "../drivingButtonStates";

describe('calculateDrivingStatus test suite', () => {
  
    describe('default driving status', () => {
      
        it('should return defaults which is not driving', () => {
            
            expect(calculateDrivingStatus()).toMatchSnapshot();
        });
    });
    
    describe('not driving at 0 speed', () => {
      
        it('should return not driving when ERROR and 0', () => {
            
            expect(calculateDrivingStatus(DrivingButtonStates.ERROR, 0)).toMatchSnapshot();
        });
    
        it('should return not driving when TURNED_OFF and 0', () => {
            
            expect(calculateDrivingStatus(DrivingButtonStates.TURNED_OFF, 0)).toMatchSnapshot();
        });
    
        it('should return not driving when DRIVE and 0', () => {
            
            expect(calculateDrivingStatus(DrivingButtonStates.DRIVE, 0)).toMatchSnapshot();
        });
    });

    describe('not driving at more than 0 speed', () => {
      
        it('should return not driving when ERROR and 1', () => {

            const throttleSpeed = 1;

            expect(calculateDrivingStatus(DrivingButtonStates.ERROR, throttleSpeed)).toMatchSnapshot();
        });

        it('should return not driving when DRIVE and 50', () => {

            const throttleSpeed = 50;

            expect(calculateDrivingStatus(DrivingButtonStates.ERROR, throttleSpeed)).toMatchSnapshot();
        });

        it('should return not driving when TURNED_OFF and 50', () => {

            const throttleSpeed = 50;

            expect(calculateDrivingStatus(DrivingButtonStates.TURNED_OFF, throttleSpeed)).toMatchSnapshot();
        });

        it('should return not driving when READY and 50', () => {

            const throttleSpeed = 50;

            expect(calculateDrivingStatus(DrivingButtonStates.READY, throttleSpeed)).toMatchSnapshot();
        });
    });
    
    describe('driving', () => {
      
        it('should drive when DRIVE and more than 0', () => {
            
            const throttleSpeed = 50;

        expect(calculateDrivingStatus(DrivingButtonStates.DRIVE, throttleSpeed)).toMatchSnapshot();
        });
    });
    
});
