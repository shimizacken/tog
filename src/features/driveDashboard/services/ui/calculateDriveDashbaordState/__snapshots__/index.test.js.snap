// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateDrivingStatus test suite default driving status should return defaults which is not driving 1`] = `false`;

exports[`calculateDrivingStatus test suite driving should drive when DRIVE and more than 0 1`] = `true`;

exports[`calculateDrivingStatus test suite not driving at 0 speed should return not driving when DRIVE and 0 1`] = `false`;

exports[`calculateDrivingStatus test suite not driving at 0 speed should return not driving when ERROR and 0 1`] = `false`;

exports[`calculateDrivingStatus test suite not driving at 0 speed should return not driving when TURNED_OFF and 0 1`] = `false`;

exports[`calculateDrivingStatus test suite not driving at more than 0 speed should return not driving when DRIVE and 50 1`] = `false`;

exports[`calculateDrivingStatus test suite not driving at more than 0 speed should return not driving when ERROR and 1 1`] = `false`;

exports[`calculateDrivingStatus test suite not driving at more than 0 speed should return not driving when READY and 50 1`] = `false`;

exports[`calculateDrivingStatus test suite not driving at more than 0 speed should return not driving when TURNED_OFF and 50 1`] = `false`;
